File: Statistics/Lesson-2/OnlineMeanCSharp/OnlineMeanCSharp/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace OnlineMeanCSharp
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            timer1.Start();
        }

        private void button2_Click(object sender, EventArgs e)
        {
            timer1.Stop();
        }

        private void timer1_Tick(object sender, EventArgs e)
        {
            
            int grade = rnd.Next(18, 31);
            count = count + 1;
            string exam = "Exam" + count.ToString() + " ";
            mean = mean + (grade - mean) / count;
            richTextBox1.AppendText(exam + grade + " ===> Current Mean:" + mean + "\n");

        }
    }
}

File: Statistics/Lesson-2/OnlineMeanVB/OnlineMeanVB/Form1.vb
Public Class Form1

    Public RandomNum As New Random

    Private Sub Button1_Click(sender As Object, e As EventArgs) Handles Button1.Click

        Me.Timer1.Start()

    End Sub

    Public CurrentMean As Double = 0

    Public count As Integer

    Public Sub Timer1_Tick(sender As System.Object, e As System.EventArgs) Handles Timer1.Tick

        count = count + 1

        Dim Exam = "Exam" & count

        Dim Grade As Integer = Me.RandomNum.Next(18, 31)

        CurrentMean = CurrentMean + (Grade - CurrentMean) / count

        Me.RichTextBox1.AppendText(Exam.PadRight(11) & Grade &
                                   " ===> Current Mean:" & CurrentMean & vbCrLf)

    End Sub

    Private Sub Button2_Click(sender As Object, e As EventArgs) Handles Button2.Click

        Me.Timer1.Stop()


    End Sub

End Class
